client/src/pages/editor.js

```diff
@@ -49,15 +75,23 @@ export class Editor {
         if (response.ok) {
           return response.json();
         }
-
-        return Promise.reject();
+        throw response;
       })
       .then(schema => {
         this.fullSchema = schema;
-        this.schema = getSchemaForSchemaEditor(schema);
-        if (schema.properties.hasOwnProperty('options')) {
-          this.optionsSchema = schema.properties.options;
+        const schemaForEditor = getSchemaForSchemaEditor(this.fullSchema);
+        this.schema = getTranslatedSchema(schemaForEditor, routeParams.tool, this.i18n);
```

Comment:
Could you put the following four lines of code to get the translated schema and options schema into a method and use it here and in the event bus?